scribble-doc/scribblings/scribble/manual.scrbl

```diff
@@ -2250,44 +2250,112 @@ correspond to the documented name of the binding and the primary
 modules that export the documented name (but this list is not
 exhaustive, because new modules can re-export the binding).}
 
+@defstruct[(exported-index-desc* exported-index-desc) ([extras (hash/dc [k symbol?]
+                                                                        [v (k)
+                                                                           (case k
+                                                                             [(kind) string?]
+                                                                             [(hidden?) boolean?]
+                                                                             [(method-name) symbol?]
+                                                                             [(constructor?) boolean?]
+                                                                             [(class-tag) tag?]
+                                                                             [else any/c])]
+                                                                        #:immutable #t)])]{
+
+Like @racket[exported-index-desc], but with additional optional
+information in an @racket[extras] hash table. All values in the ahsh
+table should be @racket[write]able in the sense that @racket[read]
+will reconstruct the value. Any symbol is allowed as a key in
+@racket[extras], but some are well-known:
+
+@itemlist[
+
+ @item{@racket['kind]: A string describing the binding's category,
+       such as @racket["procedure"] or @racket["class"]. Like most
+       other values in @racket[extras], this string is @emph{not}
+       added to the index entry's main text as shown to a user, but it
+       be rendered by some interfaces, such as to the side or in hover
+       text.}
+
+ @item{@racket['hidden?]: A boolean indicating that the index entry
+       describes a binding that is covered from a user's perspective
+       by a different index entry. These are considered redundant and
+       filtered from index rendering.}
+
+ @item{@racket['method-name]: A symbol indicating that the index entry
```

Comment:
This is only applicable when `kind` is `method`, right? It might be nice to document the invariant.